lib/fs/registry.go

```diff
@@ -0,0 +1,41 @@
+// Copyright (C) 2016 The Syncthing Authors.
+//
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this file,
+// You can obtain one at https://mozilla.org/MPL/2.0/.
+
+package fs
+
+import "errors"
+
+func init() {
+	registry[FilesystemTypeBasic] = func(root string) Filesystem {
```

Comment:
I've become somewhat skeptical of the magic registry kind of global state. I wonder if it wouldn't be cleaner to skip the registry thing and just have `NewFilesystem` explicitly know about the filesystem types we have and just `switch{}` on the `FilesystemType`